src/osd/PrimaryLogPG.cc

```diff
@@ -9803,9 +9803,9 @@ void PrimaryLogPG::sub_op_remove(OpRequestRef op)
 eversion_t PrimaryLogPG::pick_newest_available(const hobject_t& oid)
 {
   eversion_t v;
-
-  assert(pg_log.get_missing().is_missing(oid));
-  v = pg_log.get_missing().get_items().find(oid)->second.have;
+  pg_missing_item pmi;
+  assert(pg_log.get_missing().is_missing(oid, &pmi));
```

Comment:
Please do not rely on a side effect of an assert() call.  assign the result to a bool local and assert that is true.